Badges/badges_grant_add.php

```diff
@@ -41,20 +41,33 @@
     $gibbonSchoolYearID = $_GET['gibbonSchoolYearID'] ?? '';
 
     echo "<div class='linkTop'>";
-    if ($_GET['gibbonPersonID2'] != '' or $_GET['badgesBadgeID2'] != '') { echo "<a href='".$_SESSION[$guid]['absoluteURL'].'/index.php?q=/modules/Badges/badges_grant.php&gibbonPersonID2='.$_GET['gibbonPersonID2'].'&badgesBadgeID2='.$_GET['badgesBadgeID2']."'>".__('Back to Search Results').'</a>';
+
+    //Get the gibbon persion and badge IDs
+    //Set '' for safety
+    $gibbonPersonID2 = '';
+    $badgesBadgeID2 = '';
+    if (isset($_GET['gibbonPersonID2']) ||  isset($_GET['badgesBadgeID2'])) {
+        if($_GET['gibbonPersonID2'] != '' && $_GET['badgesBadgeID2'] != '')
+        {
+            //Only assign variable when it exists
+            $gibbonPersonID2 = $_GET['gibbonPersonID2'];
+            $badgesBadgeID2 = $_GET['badgesBadgeID2'];
+            //Add a "Back to Results" link.
+            echo "<a href='".$gibbon->session->get('absoluteURL').'/index.php?q=/modules/Badges/badges_grant.php&gibbonPersonID2='.$_GET['gibbonPersonID2'].'&badgesBadgeID2='.$_GET['badgesBadgeID2']."'>".__('Back to Search Results').'</a>';
+        }
```

Comment:
This could be simplified by using:

$gibbonPersonID2 = $_GET['gibbonPersonID2'] ?? '';

Any other places in the code where this would apply?

I'll leave it with you. Let me know when it is ready.

Reply:
Absolutely correct, additionally the echo below was still using the GET variable which wasn't santised!